File: python_converted/src/core/interfaces/types/npc/goals.py
from typing import Any, Dict, List
from enum import Enum


class NPCGoal:
    id: str
    type: \'GoalType\'
    priority: float
    description: str
    status: \'GoalStatus\'
    progress: float
    deadline?: float
    dependencies: List[str]
    subgoals: List[str]
    requirements: Dict[str, Any][]
    relationships?: {
      npcId: str
      minValue: float
    }[]
    conditions?: {
      type: str
      value: Any
    }[]
  }
  rewards?: {
    reputation?: float
    resources?: {
      type: str
      amount: float
    }[]
    relationships?: {
      npcId: str
      change: float
    }[]
  }
  completion: Dict[str, Any]
}
class GoalType(Enum):
    PERSONAL = 'personal'
    PROFESSIONAL = 'professional'
    SOCIAL = 'social'
    ECONOMIC = 'economic'
    FACTION = 'faction'
    QUEST = 'quest'
class GoalStatus(Enum):
    ACTIVE = 'active'
    COMPLETED = 'completed'
    FAILED = 'failed'
    ABANDONED = 'abandoned'
    BLOCKED = 'blocked'
    DEFERRED = 'deferred' 

File: src/core/interfaces/types/npc/goals.ts
export interface NPCGoal {
  id: string;
  type: GoalType;
  priority: number;
  description: string;
  status: GoalStatus;
  progress: number;
  deadline?: number;
  dependencies: string[];
  subgoals: string[];
  requirements: {
    resources?: {
      type: string;
      amount: number;
    }[];
    skills?: {
      skillId: string;
      minLevel: number;
    }[];
    relationships?: {
      npcId: string;
      minValue: number;
    }[];
    conditions?: {
      type: string;
      value: any;
    }[];
  };
  rewards?: {
    reputation?: number;
    resources?: {
      type: string;
      amount: number;
    }[];
    relationships?: {
      npcId: string;
      change: number;
    }[];
  };
  completion: {
    timeCompleted?: number;
    outcome?: string;
    achievementScore?: number;
  };
}

export enum GoalType {
  PERSONAL = 'personal',
  PROFESSIONAL = 'professional',
  SOCIAL = 'social',
  ECONOMIC = 'economic',
  FACTION = 'faction',
  QUEST = 'quest'
}

export enum GoalStatus {
  ACTIVE = 'active',
  COMPLETED = 'completed',
  FAILED = 'failed',
  ABANDONED = 'abandoned',
  BLOCKED = 'blocked',
  DEFERRED = 'deferred'
} 